CMakeLists.txt

```diff
@@ -161,33 +179,33 @@ if(BUILD_JS)
     file(GLOB common-fonts LIST_DIRECTORIES false ${CMAKE_CURRENT_SOURCE_DIR}/services/user/CommonSys/common/fonts/*)
 
     # Builds the Common Sys Lib
-    file(GLOB common_lib_sources ${CMAKE_CURRENT_SOURCE_DIR}/services/user/CommonSys/common-lib/src/*)
-    add_custom_command(
-        OUTPUT ${CMAKE_CURRENT_SOURCE_DIR}/services/user/CommonSys/common-lib/dist/common-lib.js
-        DEPENDS ${common_lib_sources}
-        COMMAND cd ${CMAKE_CURRENT_SOURCE_DIR}/services/user/CommonSys/common-lib && yarn --mutex network && yarn build
-    )
-    add_custom_target(CommonSysLib_js
-        DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/services/user/CommonSys/common-lib/dist/common-lib.js
-    )
+    add_js_target(${COMMON_DIR} common-lib.js)
 
     foreach(ENTRY ${JS_DIRS})
-        string(REGEX REPLACE "^([^:]+):([^:]+)$" \\1 PATH ${ENTRY})
-        string(REGEX REPLACE "^([^:]+):([^:]+)$" \\2 DEP ${ENTRY})
-        file(GLOB_RECURSE sources ${CMAKE_CURRENT_SOURCE_DIR}/${PATH}/src/*)
-        add_custom_command(
-            OUTPUT ${CMAKE_CURRENT_SOURCE_DIR}/${PATH}/dist/index.html
-            DEPENDS ${sources} ${common} ${common-fonts} CommonSysLib_js
-            COMMAND cd ${CMAKE_CURRENT_SOURCE_DIR}/${PATH} && yarn --mutex network && yarn build
-        )
-        add_custom_target(
-            ${DEP}
-            DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/${PATH}/dist/index.html
-        )
+        add_js_target(${ENTRY} index.html ${common} ${common-fonts} CommonSysLib_js)
     endforeach()
-endif()
 
-if(BUILD_JS)
+    set(byproducts ${CMAKE_CURRENT_SOURCE_DIR}/services/user/AdminSys/ui/wasm-psibase)
+
+    add_js_target(${ADMIN_DIR} index.html
+        ${byproducts}/wasm-psibase_bg.js
+        ${byproducts}/wasm-psibase_bg.wasm
+        ${byproducts}/wasm-psibase_bg.wasm.d.ts
+        ${byproducts}/wasm-psibase.d.ts
+        ${byproducts}/wasm-psibase.js
+    )
+
+    add_custom_target(wasm-psibase ALL
+    COMMAND wasm-pack build ${CMAKE_CURRENT_SOURCE_DIR}/rust/psibase --out-dir ${byproducts} --out-name wasm-psibase
```

Comment:
I'm seeing intermediate build products created in the source tree rather than in the build tree. I'm not sure whether this is fixable, because wasm-pack seems to hard-code the path.